### ISSUE TITLE
App crashing mid way through operating

### ISSUE THREAD

**Issue by SuperSandro2000** [d4rken/gplay-batchtool#2]
App randomly crashes after 5 to 15 removed apps. 
S8 Oreo, Swift Installer
Log file: [2018-07-03-00-22-40.txt](https://github.com/d4rken/gplay-batchtool/files/2156984/2018-07-03-00-22-40.txt)

**Comment by d4rken** [comment]
Hm the beginning of the crash seems to be missing, I can't see the actual exception in the log, can you?

What's SwiftInstaller?

**Comment by SuperSandro2000** [comment]
There was no exception in the log. It didn't even show up under the correct log level.

Swift installier is a theme installer comparable to substratum and paid.
https://play.google.com/store/apps/details?id=com.brit.swiftinstaller

**Comment by d4rken** [comment]
Sure? I mean this looks like a stacktrace, just with the first part missing.

```
07-03 00:21:54.776 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.Crawler.doFindAll(Crawler.java:57)
07-03 00:21:54.776 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.Crawler.findAll(Crawler.java:36)
07-03 00:21:54.777 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.Crawler$Builder.lambda$findAll$0$Crawler$Builder(Crawler.java:123)
07-03 00:21:54.777 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.Crawler$Builder$$Lambda$0.subscribe(Unknown Source:2)
07-03 00:21:54.783 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.ScanService.removeEntries(ScanService.java:212)
07-03 00:21:54.783 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.ScanService.lambda$onStartClicked$4$ScanService(ScanService.java:107)
07-03 00:21:54.783 W/System.err(12847): 	at eu.darken.gplaybatchtool.main.core.ScanService$$Lambda$4.subscribe(Unknown Source:2)
```

Hm the last line we get is this
https://github.com/d4rken/gplay-batchtool/blob/314f2a09bd737dee1f059979ee5e62a18a28278d/app/src/main/java/eu/darken/gplaybatchtool/main/core/Crawler.java#L57

My initial hunch would be that the cause is `current.getChild(i)`.
Possibly an `IllegalArgumentException` or `IndexOutOfBoundsException` depending on whether `queue` allows null to be added.

**Comment by SuperSandro2000** [comment]
I tried it again a few minutes back and can't reproduce it. The App cleared around 30 items without any problems. Maybe I had YouTube pop-up open, night filter on or needed to reboot my phone.

**Comment by d4rken** [comment]
If someone is up for it we could add a defensive check.

**Comment by d4rken** [comment]
Fixed in [v0.6](https://github.com/d4rken/gplay-batchtool/releases/tag/v0.6). :+1:
